### ISSUE TITLE
Discuss the API for a ScreenBuilder to allow a stable API for building in-game screens

### ISSUE THREAD

**Issue by SRAZKVT** [OrnitheMC/ornithe-standard-libraries#1]
Let's discuss the API such a builder should have, including, but not limited to : 
- Requirements
- Classes
- Methods
- Arguments

**Comment by SpaceWalkerRS** [comment]
I reckon it makes sense to keep it fairly simple. So e.g. no complex structures like sub-screens (just open a new screen like vanilla does). Though ig we don't have to lock it down, so people can extend the system with more complex structures if they wish.

**Comment by SRAZKVT** [comment]
yes, it being extandable should be a requirement, probably using minimal interfaces for most basic components. Though, i guess common extensions making it into the base builder wouldn't hurt, if it is something very commonly used, but we aren't there yet.
